stl/inc/type_traits

```diff
@@ -1665,12 +1665,24 @@ struct _Invoker_ret<_Unforced, false> { // selected for _Rx being _Unforced
 #pragma warning(disable : 4244) // 'argument': conversion from '_From' to '_To', possible loss of data
 #pragma warning(disable : 4365) // 'argument': conversion from '_From' to '_To', signed/unsigned mismatch (/Wall)
 #pragma warning(disable : 5215) // '%s' a function parameter with a volatile qualified type is deprecated in C++20
+
+#ifdef __clang__
+#pragma clang diagnostic push

```

Comment:
do we want a transition tag here for when we remove volatile from our headers?

Reply:
We generally use `TRANSITION` to mark code that should be changed when compiler bugs are fixed, or compiler features are implemented. Here, I don't believe that we could remove these warning suppressions unless and until the Standard changes to forbid `volatile` inputs into this machinery (e.g. nobody should be able to ask whether something is nothrow implicitly convertible to `volatile X`). That might happen in C++23, or maybe never. In any event, it would be changed by a paper (or possibly an LWG issue), which we already have systems for tracking, so I don't believe that `TRANSITION` comments would achieve anything here.